crates/rolldown_common/src/types/js_regex.rs

```diff
@@ -8,10 +8,28 @@ pub enum HybridRegex {
 }
 
 impl HybridRegex {
-  pub fn new(source: &str) -> anyhow::Result<Self> {
-    match regex::Regex::new(source).map(HybridRegex::Optimize) {
+  pub fn new(pattern: &str) -> anyhow::Result<Self> {
+    match regex::Regex::new(pattern).map(HybridRegex::Optimize) {
       Ok(reg) => Ok(reg),
-      Err(_) => regress::Regex::new(source).map(HybridRegex::Ecma).map_err(anyhow::Error::from),
+      Err(_) => regress::Regex::new(pattern).map(HybridRegex::Ecma).map_err(anyhow::Error::from),
+    }
+  }
+
+  pub fn with_flags(pattern: &str, flags: &str) -> anyhow::Result<Self> {
+    let mut result = String::new();
+    for c in flags.bytes() {
+      match c {
+        b'i' | b'm' | b's' | b'u' | b'R' | b'U' | b'x' => result.push(c as char),
+        _ => {}
+      }
+    }
+
+    let regex_pattern = if result.is_empty() { pattern } else { &format!("(?{result}){pattern}") };
```

Comment:
```suggestion
    let regex_pattern = if flags.is_empty() { pattern } else { &format!("(?{flags}){pattern}") };

```

Reply:
The extra allocation is not necessary, we should fallback to `regress` and even raise an error if the `flags` is not supported.